rbd/watchers_mimic_test.go

```diff
@@ -0,0 +1,115 @@
+// +build !luminous
+//
+// Ceph Mimic is the first version that supports watchers through librbd.
+
+package rbd
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestListWatchers(t *testing.T) {
+	conn := radosConnect(t)
+	require.NotNil(t, conn)
+	defer conn.Shutdown()
+
+	poolname := GetUUID()
+	err := conn.MakePool(poolname)
+	require.NoError(t, err)
+	defer conn.DeletePool(poolname)
+
+	ioctx, err := conn.OpenIOContext(poolname)
+	require.NoError(t, err)
+	defer ioctx.Destroy()
+
+	name := GetUUID()
+	options := NewRbdImageOptions()
+	err = CreateImage(ioctx, name, 1<<22, options)
+	require.NoError(t, err)
+
+	var image, image2, image3 *Image
```

Comment:
What I have an issue with here is that allowing the objects to be referenced outside the sub-tests means that the sub tests are dependent on each other. They should only depend on the state established at the outer function. You can ask go test to only run one sub-test and that shouldn't break.
I'm not saying image shouldn't be defined outside the sub tests just that if defined outside the subtests all mutation of the state should be done outside the subtests.  I'm willing to do a peer-to-peer session if you want to if I'm not being clear in these reviews.

Reply:
Right, at the moment the tests depend on eachother. I'll structure them a little different and make them self-contained.